File: jdr-final/src/main/java/com/monapp/entity/Cellule.java
package com.monapp.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonView;

import lombok.Getter;
import lombok.Setter;

@Entity
@Setter
@Getter
public class Cellule {

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="id_cellule")
	@JsonView(Views.Common.class)private int id;
	
	@Column
	@NotNull
	@JsonView(Views.Common.class)
	private String sceneDescription;
	
	@OneToOne(fetch=FetchType.EAGER)
	@JoinColumn(name="routesPossible")
	@JsonView(Views.Cellule.class)
	private Route routesPossible; 
	
	@ManyToOne
    @JoinColumn(name="plateau")
    private Plateau plateau;
	
	@Column
	@JsonView(Views.Common.class)
	private String imageLien;
	
	@Column
	@NotNull
	@JsonView(Views.Common.class)
	private int coordoneeX;
	
	@Column
	@NotNull
	@JsonView(Views.Common.class)
	private int coordoneeY;

	@Column
	@JsonView(Views.Common.class)
	private int position;
	
	@Column
	@JsonView(Views.Common.class)
	private int numeroDeCase;
	
	public Cellule() {}
	public Cellule(String sceneDescription, Route routesPossible, String imageLien, int coordonneX, int coordonneY, int position, int numeroDeCase) {
		this.sceneDescription = sceneDescription;
		this.routesPossible = routesPossible;
		this.imageLien = imageLien;
		this.coordoneeX = coordonneX;
		this.coordoneeY = coordonneY;
		this.position = position;
		this.numeroDeCase = numeroDeCase;
	}
	@Override
	public String toString() {
		return "Cellule [id=" + id + ", sceneDescription=" + sceneDescription + ", imageLien="
				+ imageLien + ", coordoneeX=" + coordoneeX + ", coordoneeY=" + coordoneeY + ", position=" + position
				+ ", numeroDeCase=" + numeroDeCase + "]";
	}
	
	public void addRoute(Route route) {
		route.addCellule(this);
		this.setRoutesPossible(route);
	}
	
	
	
}

File: app/cellule.ts
import {Route} from './route';
import {Plateau} from './plateau';

export class Cellule {

  constructor(  private _id: number,
                private _plateau: Plateau,
                private _RoutePossible: Route,
                private _imageLien: string,
                private _position: number,
                private _coordonneX: number,
                private _coordonneY: number) {}


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get plateau(): Plateau {
    return this._plateau;
  }

  set plateau(value: Plateau) {
    this._plateau = value;
  }

  get RoutePossible(): Route {
    return this._RoutePossible;
  }

  set RoutePossible(value: Route) {
    this._RoutePossible = value;
  }

  get imageLien(): string {
    return this._imageLien;
  }

  set imageLien(value: string) {
    this._imageLien = value;
  }

  get position(): number {
    return this._position;
  }

  set position(value: number) {
    this.position = value;
  }

  get coordonneX(): number {
    return this._coordonneX;
  }

  set coordonneX(value: number) {
    this._coordonneX = value;
  }

  get coordonneY(): number {
    return this._coordonneY;
  }

  set coordonneY(value: number) {
    this._coordonneY = value;
  }
}
